pennylane/_device.py

```diff
@@ -340,6 +341,9 @@ def map_wires(self, wires):
         Returns:
             Wires: wires with new labels
         """
+        if wires in self._cached_wires:
+            return self._cached_wires[wires]
+
```

Comment:
Nice solution for now. When we are refactoring the device class (if this ever happens :)  I will help turning all the current patches into a runtime-conscious solution overall!